mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -231,4 +241,57 @@ public void searchShouldNotReturnResultsFromOtherUsers() throws Exception {
         assertThat(metaDatas.get(0).getPath()).isEqualTo(MailboxPath.inbox(session));
     }
 
+    @ContractTest
+    public void updateAnnotationsShouldUpdateStoredAnnotation() throws MailboxException {
+        MailboxPath inbox = setUpMailbox();
```

Comment:
You should at least duplicate `Assume.assumeTrue(mailboxManager.hasCapability(MailboxCapabilities.Annotation));`
For the "I still prefer to separate duplicated code to a function" part of your answer : duplication can be a good thing in the testing context, readability is different in test and application code. We expect to understand a test without too many jumps around, to not contains abstraction, no control flow instruction, etc. We usually duplicate things to have everything under our eyes and be sure nothing gets done just because we had a common method to do it (something that becomes common when you start extracting code in common methods).